tasks/task_manager.hh

```diff
@@ -344,12 +354,22 @@ public:
             id = task->id();
 
             if (parent_d) {
-                co_await _tm.container().invoke_on(parent.get_owner_shard(), [task = std::move(parent), child = std::move(task)] (task_manager& tm) mutable {
-                    task->add_child(std::move(child));
-                });
+                co_await add_child(std::move(parent), std::move(task));
             }
             co_return id;
         }
+
+        // Responsibility for task_impl_ptr correctness is put on caller.
```

Comment:
I don't understand what does that responsibility for correctness mean.
Please be more specific about what the caller is responsible for.